packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -0,0 +1,64 @@
+import { StreamID } from '@ceramicnetwork/streamid'
+import { AnchorRequestStoreListResult } from '../store/anchor-request-store.js'
+import { AnchorStatus, DiagnosticsLogger, LogStyle, SyncOptions } from '@ceramicnetwork/common'
+import { Repository } from './repository.js'
+import PQueue from 'p-queue'
+
+const RESUME_QUEUE_CONCURRENCY = 30
+const RESUME_BATCH_SIZE = RESUME_QUEUE_CONCURRENCY * 5
+
+export class AnchorResumingService {
+  /**
+   * Resume running states from anchor request store
+   */
+  readonly resumeQ: PQueue
+
+  /**
+   * true iff the repository is in the process of closing
+   *
+   * @private
+   */
+  #shouldBeClosed = false
+
+  constructor(private readonly logger: DiagnosticsLogger) {
+    this.resumeQ = new PQueue({ concurrency: RESUME_QUEUE_CONCURRENCY })
+  }
+
+  async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
+    this.#shouldBeClosed = false
```

Comment:
is this line necessary?  It makes me think that shouldBeClosed could sometimes go from false to true back to false again.

Reply:
If the only way to get this running again after calling `close()` is to create a new instance, then it's not necessary. I'll remove it.